app/src/lib/stores/pull-request-store.ts

```diff
@@ -326,22 +326,29 @@ export class PullRequestStore extends TypedBaseStore<GitHubRepository> {
 
     const table = this.pullRequestDatabase.pullRequests
     const prsToInsert = new Array<IPullRequest>()
-    let githubRepo: GitHubRepository | null = null
 
     for (const pr of pullRequestsFromAPI) {
-      // Once the repo is found on first try, no need to keep looking
-      if (githubRepo == null && pr.head.repo != null) {
-        githubRepo = await this.repositoryStore.upsertGitHubRepository(
-          repository.endpoint,
-          pr.head.repo
+      // `pr.head.repo` represents the source of the pull request. It might be
+      // a branch associated with the current repository, or a fork of the
+      // current repository.
+      //
+      // In cases where the user has removed the fork of the repository after
+      // opening a pull request, this can be `null`, and the app will not store
+      // this pull request.
```

Comment:
We can track this as a `tech-debt` task for someone to investigate, and if we can find a public repository with an open PR where the fork has been removed, all the better for testing...

Reply:
Here you go:

 - https://github.com/shiftkey/vcr-sharp/pull/13
 - https://api.github.com/repos/shiftkey/vcr-sharp/pulls/13

Reply:
Additional notes:

The URL for this problem PR is `x-github-client://openRepo/https://github.com/shiftkey/vcr-sharp?branch=shiftkey-tester-patch-1` which doesn't actually exist:

```shellsession
$ git for-each-ref refs/remotes/origin
9842b3f6144c3e44ac1af0d419b55e9ee114831c commit	refs/remotes/origin/HEAD
9842b3f6144c3e44ac1af0d419b55e9ee114831c commit	refs/remotes/origin/master

$ git ls-remote origin
9842b3f6144c3e44ac1af0d419b55e9ee114831c	HEAD
9842b3f6144c3e44ac1af0d419b55e9ee114831c	refs/heads/master
6b732b6032e05ce0365fc23435e834e66d3bf28e	refs/pull/1/head
b6e0d926c527aaeaee5cc2c545728a55d95ba87e	refs/pull/10/head
4af3fff362b647059c285d7feac14041038aaa89	refs/pull/11/head
b2480c6084e02580cfab638f2e28d224149e7915	refs/pull/12/head
4eab45b996fa7365a7a62750f28c4b0ff4dfa3f0	refs/pull/13/head
733e39ec85507c4dbf8413f0da768b31813b8a8a	refs/pull/13/merge
7ff8829aa34a8ae0baafe73a264fcde1ea3f4f76	refs/pull/4/head
00feb00576633447a0fab3bed57be4a590604578	refs/pull/6/head
c7be263af918299d9da4beb85153bbfbe437eb6f	refs/pull/7/head
600aac5ad846f6bce7c7891787af2c362bde3fe8	refs/pull/9/head
```

And Desktop will error out if you try and click into it:

<img width="533" alt="screen shot 2018-02-12 at 5 21 52 pm" src="https://user-images.githubusercontent.com/359239/36085947-465a4502-1019-11e8-9474-0638543fa70c.png">


The instructions for manually merging in this PR are _interesting_:

<img width="685" src="https://user-images.githubusercontent.com/359239/36085863-bfc84476-1018-11e8-98ef-42e2628d7d7b.png">

The hypothetical workflow in the app could look like this:

 - warn the user that the PR came from an unknown repository, and the app will try and apply this manually to `master` if they wish to continue
 - create a new branch off `master` and name it using some sort of conventions
 - fetch the diff from the `master` and apply it using `git am -3`
 
At this point I'm not sure if the changes will be committed, but we may need to handle merge conflicts (because the changes don't apply cleanly) and I think this is all worth punting to post 1.1.

Reply:
Is there any way to fetch the `head` ref from GitHub, then check it out into a branch?

Reply:
@j-f1 you can do it like this:

```
$ git fetch origin pull/13/head:shiftkey-tester-patch-1
remote: Counting objects: 3, done.
remote: Total 3 (delta 2), reused 2 (delta 2), pack-reused 1
Unpacking objects: 100% (3/3), done.
From https://github.com/shiftkey/vcr-sharp
 * [new ref]         refs/pull/13/head -> shiftkey-tester-patch-1
```